warehouse/templates/manage/token.html

```diff
@@ -0,0 +1,96 @@
+{#
+ # Licensed under the Apache License, Version 2.0 (the "License");
+ # you may not use this file except in compliance with the License.
+ # You may obtain a copy of the License at
+ #
+ # http://www.apache.org/licenses/LICENSE-2.0
+ #
+ # Unless required by applicable law or agreed to in writing, software
+ # distributed under the License is distributed on an "AS IS" BASIS,
+ # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ # See the License for the specific language governing permissions and
+ # limitations under the License.
+-#}
+{% extends "manage_base.html" %}
+
+{% set user = request.user %}
+{% set title = "API token management" %}
+
+{% block account_active %}vertical-tabs__tab--is-active{% endblock %}
+{% block account_mobile_active %}vertical-tabs__tab--is-active{% endblock %}
+
+{% block title %}{{ title }}{% endblock %}
+
+{% block main %}
+  <h1 class="page-title">{{ title }}</h1>
+
+  {% if serialized_macaroon %}
+  <section id="provisioned-key">
+    <div class="callout-block callout-block--warning">
+      <p>
+        This token will only appear once. Copy it now.
+      </p>
+      <!-- TODO: This needs to be word-wrapped -->
+      <p>
+        {{ serialized_macaroon }}
+      </p>
+    </div>
+  </section>
+  {% endif %}
+
+  {% if user.macaroons|length > 0 %}
+  <section id="active-tokens">
+    <h2>Active API tokens</h2>
+    <table class="table table--light">
+      <thead>
+        <th>Description</th>
+        <th>Identifier</th>
+        <th>Created</th>
+        <th colspan="2">Last used</th>
+      </thead>
```

Comment:
![table-and-api-key-for-github-review](https://user-images.githubusercontent.com/842790/61420213-7ff57480-a8cf-11e9-9cc3-a58e6354de20.png)

Is it possible for one of the headings to be **Scope**? If not, could we add some text (next to the "copy it now") also reminding the user to make a note of the scope?

Reply:
I'm also curious -- maybe I've missed something -- about what the "Identifier" (hash?) is for. It's clearly a unique identifier for the API key, but is the user ever going to need to refer to it, either in her own systems or when communicating with PyPI tech support or something like that?

Reply:
Yep, I was giving some thought to how best to represent the token's scope -- it can currently be a single user or a set of packages. That's relatively easy to present, but future changes to the caveat language might require more complex presentations.

We can punt that down the road though and just handle the user + package set cases for now, though.

Reply:
Oh, and re: the identifier: macaroons require a unique identifier, but users definitely don't need to see it. It was there for my debugging purposes; I'll remove it 🙂

Reply:
One possible benefit of exposing it is to make it easier for a user to figure out what row in their tokens corresponds with a token they happen to have possession of.

Of course, there are other ways of doing this, like embedding the description into the macaroon identifier as well.

Reply:
Yep, that's a good point. We don't currently embed the description into the macaroon at all (and, at 100 characters, doing so might increase the length quite a bit), so I'll re-add the identifier.

Reply:
Here's what the scope looks like with just the dictionary dumped into it:

<img width="853" alt="Screen Shot 2019-07-18 at 1 03 57 PM" src="https://user-images.githubusercontent.com/3059210/61477119-94338300-a95c-11e9-9297-f6f49397140d.png">

Reply:
> One possible benefit of exposing it is to make it easier for a user to figure out what row in their tokens corresponds with a token they happen to have possession of.

So the imagined user here is storing this structured data (description, identifier, and token) locally, in a config file? Just checking!

Reply:
> So the imagined user here is storing this structured data (description, identifier, and token) locally, in a config file?

The token has the identifier embedded in it, so the user would probably only store the token. The description is only stored in the DB for presentation in the web interface -- it isn't tightly coupled to the token itself. We could change that and embed it into the token directly, but at the cost of longer tokens and (IMO) relatively little benefit.